File: C++/Problems_Array/Easy/largest_element.cpp
/*
Largest element of an array
*/

#include <iostream>
using namespace std;
int largest_element(int arr[], int n){
    int largest = arr[0];
    for(int i = 0; i<n; i++){
        if (arr[i] >= largest){
            largest = arr[i];
        }
    }
    return largest;
}

int main(){
    int n= 5;
    int arr[5];
    for(int i = 0; i <n; i++){

        cin>> arr[i];
    }
    int greatest_elemnt = largest_element(arr, n);
    cout<<"The largest element of the input array is "<< greatest_elemnt;
    return 0;
}

File: Python/Problems_Array/Easy/Largest_element.py
"""_summary_: Largest element in Array
"""

def largest_element(input_array):
    # But time complexity of this is O(N)--> Can we imporve it?
    largest = input_array[0]
    for i in range(len(input_array)):
        if input_array[i] >= largest:
            largest = input_array[i]
        else:
            pass

    return largest



if __name__ == "__main__":
    input_array = [2, 5,10, 54, 0, 1]
    greatest_element = largest_element(input_array=input_array)
    print("Greatest element in the input array is : ", greatest_element)
